packages/core-data/src/entity-types/term.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * Internal dependencies
+ */
+import type { Context, ContextualField, OmitNevers } from './helpers';
+
+import type { BaseEntityRecords as _BaseEntityRecords } from './base-entity-records';
+
+declare module './base-entity-records' {
+	export namespace BaseEntityRecords {
+		export interface Term< C extends Context > {
+			/**
+			 * A human-readable description of the term.
+			 */
+			description: ContextualField< string, 'view' | 'edit', C >;
+			/**
+			 * The title for the term.
+			 */
+			name: string;
+			/**
+			 * An alphanumeric identifier for the term.
+			 */
+			slug: string;
+			/**
+			 * The number of objects with the term.
+			 */
+			count: number;
+			/**
+			 * The ID for the term.
+			 */
+			id: number;
+			/**
+			 * The link for the term.
+			 */
+			link: string;
+			/**
+			 * Meta fields for the term.
+			 */
+			meta: { [ key: string ]: unknown };
```

Comment:
Isn't meta essentially an associative array of meta fields? And because of the database schema, they'll always end up strings?

Also it supports `view` and `edit` contexts, right?

```suggestion
			meta: ContextualField<
				Record< string, string >,
				'view' | 'edit',
				C
			>;
```